crates/solvers/src/infra/contracts.rs

```diff
@@ -9,12 +9,12 @@ pub struct Contracts {
 }
 
 impl Contracts {
-    pub fn for_chain(chain: eth::ChainId) -> Self {
+    pub fn for_chain(chain: network::Network) -> Self {
         let a = |contract: &contracts::ethcontract::Contract| {
             eth::ContractAddress(
                 contract
                     .networks
-                    .get(chain.network_id())
+                    .get(&chain.chain_id().to_string())
```

Comment:
Not sure what others think, but to me personally, `chain` would be a more appropriate name for the new enum since, IMO, `network` is too generic.

Reply:
I'm fine with chain. But then `Network::chain_id()` should be renamed to `Chain::id()` as well.